test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -680,10 +685,16 @@ public void afterStart() {
         return new NodeAndClient(name, node, nodeId);
     }
 
+    private String getNodePrefix(Settings settings) {
+        return nodePrefix + getRoleSuffix(settings);
+    }
+
+    private int getNodeOrdinal(String nodeName, Settings settings) {
```

Comment:
I don't really like this. Do we not know the node ordinal in all the places that `buildNode` is called?

Reply:
Yes, I agree it's ugly. I thought without it, the code would be even more ugly, however, seems that I've managed to get rid of it in this commit https://github.com/elastic/elasticsearch/pull/36559/commits/91bb6e98aef15ef9299b27905efe83231a8446d5